Update to Piggieback 0.1.2 and add support for cljs.repl compiler environments

dzacarias opened the pull request:
I was getting NullPointerExceptions while trying to use cljs-noderepl with the latest ClojureScript. This would also happen with Piggieback/nREPL.

After updating to the latest Piggieback, I realised that the call to `node-setup` in `nrepl-env` was causing problems. It would give a NullPointerException while internally deref'ing the compiler var.

I've now wrapped the node-setup function to use the repl-env compiler environment or the default one. I also added the default compiler environment to the node repl-env.

With these changes, the REPL can now be launched using the 3 prescribed methods in the README, using the latest ClojureScript, Piggieback and Node.

Commit:
Bump piggieback version to 0.1.2

Commit:
Since version 0.1.0, Piggieback no longer requires an explicit
setup prior to
cemerick.piggieback/cljs-repl (c.f. https://github.com/cemerick/piggieback/blob/c6d1406eb2b21cfa74e53521a293d91641b375c0/CHANGES.md#010
)

This was causing problems while starting a new session with run-node-nrepl

Commit:
Bump version to 0.1.11

Commit:
Changed working version to 0.1.11-SNAPSHOT, for development. Whenever
this gets merged to the upstream repo and released, it should be changed
to the release version
Also changed the README to point to the proper current version

Commit:
Finally figured out (and fixed) what was wrong when using the latest
Piggieback (0.1.2) and ClojureScript.
There's now a compiler environment that should be set before evaluating
forms. I've now added it to the setup function and to the repl-env. It
uses cljs.repl/default-compiler-env

bodil commented:
So... do you still want me to merge this, or should it be closed? Not entirely sure at this point. :)

dzacarias commented:
Yes, I want to merge it. Sorry for the mess. I'm very new to Clojure(Script) and didn't want to mess anything up. It's been working for me in development. Here's a summary of the (final) merge: 
- Added a default compiler env before evaluating, so it works with recent ClojureScript versions
- Moved the version to 0.1.11-SNAPSHOT, so you can cut a new release whenever you want to.

DjebbZ commented:
Same problem when following the instructions, I get NullPointer Exception when trying to run the node-nrepl from the REPL `NullPointerException   clojure.core/deref-future (core.clj:2108)`. So if this PR fixes the problem, I'd be happy to see it it merged :) Pretty sure the noderepl's nrepl can help integration between Light Table, ClojureScript and Node.js.

bodil commented:
Merged, fresh release incoming.

dzacarias commented:
Great, thanks!
